Sources/NIOTransportServices/NIOTSConnectionChannel.swift

```diff
@@ -177,37 +179,46 @@ internal final class NIOTSConnectionChannel {
 
     /// Whether a call to NWConnection.receive has been made, but the completion
     /// handler has not yet been invoked.
-    private var outstandingRead: Bool = false
+    internal var outstandingRead: Bool = false
 
     /// The options for this channel.
-    private var options: ConnectionChannelOptions = ConnectionChannelOptions()
+    internal var options = TransportServicesChannelOptions()
 
     /// Any pending writes that have yet to be delivered to the network stack.
-    private var pendingWrites = CircularBuffer<PendingWrite>(initialCapacity: 8)
+    internal var pendingWrites = CircularBuffer<PendingWrite>(initialCapacity: 8)
 
     /// An object to keep track of pending writes and manage our backpressure signaling.
-    private var backpressureManager = BackpressureManager()
+    internal var _backpressureManager = BackpressureManager()
 
     /// The value of SO_REUSEADDR.
-    private var reuseAddress = false
+    internal var reuseAddress = false
 
     /// The value of SO_REUSEPORT.
-    private var reusePort = false
+    internal var reusePort = false
     
     /// The value of the allowLocalEndpointReuse option.
-    private var allowLocalEndpointReuse = false
+    internal var allowLocalEndpointReuse = false
 
     /// Whether to use peer-to-peer connectivity when connecting to Bonjour services.
-    private var enablePeerToPeer = false
+    internal var enablePeerToPeer = false
 
     /// The default multipath service type.
-    private var multipathServiceType = NWParameters.MultipathServiceType.disabled
+    internal var multipathServiceType = NWParameters.MultipathServiceType.disabled
 
     /// The cache of the local and remote socket addresses. Must be accessed using _addressCacheLock.
     private var _addressCache = AddressCache(local: nil, remote: nil)
 
+    internal var addressCache: AddressCache {
```

Comment:
Is it necessary for us to keep this computed property rather than exposing the stored one?